monads/src/FailCont/Main.hs

```diff
@@ -9,11 +9,31 @@ data Error
   | DivisionByZero
   deriving (Show, Eq)
 
+parseInput :: String -> FailCont r Error Int
+parseInput "" = toFailCont $ Left EmptyInput
+parseInput s  = toFailCont $ maybe (Left (ParseFailed s)) Right (readMaybe s)
+
 addInts :: String -> String -> FailCont r Error Int
-addInts = undefined 
+addInts x y = do
+  a <- parseInput x
+  b <- parseInput y
+  return (a + b)
+
+divInts :: String -> String -> FailCont r Error Int
+divInts x y = do
+  a <- parseInput x
+  b <- parseInput y
+  toFailCont $ if b == 0
+               then Left DivisionByZero
+               else Right (a `div` b)
 
-divInts :: String -> String -> FailCont r Error Int 
-divInts = undefined 
+sumRange :: String -> String -> FailCont r Error Int
+sumRange x y = do
+  a <- parseInput x
+  b <- parseInput y
+  toFailCont $ if a > b
+               then Left (ParseFailed "a must be less than or equal to b")
```

Comment:
This is not a parsing error.

Reply:
[Fixed](https://github.com/kajigor/fp-2024-cub-fall/pull/131/commits/c6931de04262d40b779be4ca49527031a44bcbd3#diff-bc1bf21989c636399fdcd35687ee7430e9a446cf151150ad07b9794b158cc3dd) by adding new error type